tests/DatabaseSchemaManagerTest.php

```diff
@@ -0,0 +1,143 @@
+            'database.connections.pgsql.schema' => 'public',
+            'tenancy.database.based_on' => null,
+            'tenancy.database.suffix' => '',
+            'tenancy.database.separate_by' => 'schema',
+            'tenancy.database_managers.pgsql' => \Stancl\Tenancy\TenantDatabaseManagers\PostgreSQLSchemaManager::class,
+        ]);
+    }
+
+    /** @test */
+    public function reconnect_method_works()
+    {
+        $old_connection_name = app(\Illuminate\Database\DatabaseManager::class)->connection()->getName();
+
+        tenancy()->init('test.localhost');
+
+        app(\Stancl\Tenancy\DatabaseManager::class)->reconnect();
+
+        $new_connection_name = app(\Illuminate\Database\DatabaseManager::class)->connection()->getName();
+
+        $this->assertSame($old_connection_name, $new_connection_name);
+    }
+
+    /** @test */
+    public function the_default_db_is_used_when_based_on_is_null()
+    {
+        config(['database.default' => 'pgsql']);
+
+        $this->assertSame('pgsql', config('database.default'));
+        config([
+            'database.connections.pgsql.foo' => 'bar',
+            'tenancy.database.based_on' => null,
+        ]);
+
+        tenancy()->init('test.localhost');
+
+        $this->assertSame('tenant', config('database.default'));
+        $this->assertSame('bar', config('database.connections.' . config('database.default') . '.foo'));
+    }
+
+    /** @test */
+    public function make_sure_using_schema_connection()
+    {
+        $tenant = tenancy()->create(['schema.localhost']);
+        tenancy()->init('schema.localhost');
+
+        $this->assertSame($tenant->getDatabaseName(), config('database.connections.' . config('database.default') . '.schema'));
+    }
+
+    /** @test */
+    public function databases_are_separated_using_schema_and_not_database()
+    {
+        tenancy()->create('foo.localhost');
+        tenancy()->init('foo.localhost');
+        $this->assertSame('tenant', config('database.default'));
+        $this->assertSame('main', config('database.connections.tenant.database'));
+
+        $schema1 = config('database.connections.' . config('database.default') . '.schema');
+        $database1 = config('database.connections.' . config('database.default') . '.database');
+
+        tenancy()->create('bar.localhost');
+        tenancy()->init('bar.localhost');
+        $this->assertSame('tenant', config('database.default'));
+        $this->assertSame('main', config('database.connections.tenant.database'));
+
+        $schema2 = config('database.connections.' . config('database.default') . '.schema');
+        $database2 = config('database.connections.' . config('database.default') . '.database');
+
+        $this->assertSame($database1, $database2);
+        $this->assertNotSame($schema1, $schema2);
+    }
+
+    /** @test */
+    public function schemas_are_separated()
+    {
+        // copied from DataSeparationTest
```

Comment:
Based of your comment maybe we should create a trait for these common database tests. So we are sure that al behaves the same?

Reply:
It's pretty simple code and I value tests being explicit more than I value them being clean.